core/Services/Calendar/TimeManager.vala

```diff
@@ -114,4 +114,20 @@ public class Calendar.TimeManager : Object {
 
         return instance;
     }
+
+    // Sets up a new TimeManager for testing, which uses settable values.
+    // This overwrites the default instance, so once this is called it can be
+    // get_default can be used as usual to return the test object.
+    public static unowned TimeManager setup_test (ICal.Timezone system_timezone) {
+        if (instance != null) {
+            warning ("Resetting default TimeManager to new testing instance");
+        }
+
+        instance = new TimeManager.for_testing (system_timezone);
+        return instance;
+    }
+
+    private TimeManager.for_testing (ICal.Timezone system_timezone) {
+        this.system_timezone = system_timezone;
+    }
```

Comment:
The difficulty here is that `TimeManager` (as designed, I suppose that could change) is a singleton. It's used within other functions using the `get_default` function, so just generating a `TimeManager` instance in a test won't mean that the new instance is used by any other code. Does that make sense? 

If we want to do something like you suggest here and not overwrite the default instance, I'd have to refactor all the other code that uses `TimeManager`.

Reply:
I see. Alright, I guess this can be done in a subsequent PR then.

Reply:
Alternatively, if you can think of a way to make TimeManager not a singleton, that would be great. Most of it doesn't require being a singleton really. The only issue is that since #629 we store the system time zone in it for speed. We have to share that data somehow (or get it to the places where it's relevant) without re-calculating it every time, and I don't have any great ideas for how to do that without some universal state like a singleton.